tree/treeplayer/src/TTreeFormula.cxx

```diff
@@ -363,14 +363,18 @@ void TTreeFormula::DefineDimensions(Int_t code, Int_t size,
    }
 
    Int_t vsize = 0;
+   bool scalarindex = false;
 
    if (fIndexes[code][fNdimensions[code]]==-2) {
       TTreeFormula *indexvar = fVarIndexes[code][fNdimensions[code]];
       // ASSERT(indexvar!=0);
       Int_t index_multiplicity = indexvar->GetMultiplicity();
       switch (index_multiplicity) {
-         case -1:
          case  0:
+            scalarindex = true;
+            vsize = 1;
+            break;
+         case -1:
          case  2:
             vsize = indexvar->GetNdata();
             break;
```

Comment:
This looks like a case for an `if` and/or a ternary operator. Personally, I find the fall throughs harder to read.